File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;
import entity.Result;

import java.util.List;
import java.util.Map;

public interface BrandService {
   public List<TbBrand> findAll();
   /**
    * 品牌分页查询
    */
   public PageResult findPage(TbBrand tbBrand,int pageNum,int pageSize);
   /**添加
    */
   public void save(TbBrand tbBrand);
	/**
	 * 修改
	 */
	public TbBrand findOne(Long id);
	/**
	 * 删除
	 */
	public void deleteMulti(Long[] ids);

	/**
	 * 模板管理中的select2的关联品牌的下拉选项
	 */
	public List<Map> select2OptionList();
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
app.service('brandService',function ($http) {
    this.search=function (pageNum,pageSize,searchEntity) {
        return $http.post('../brand/search.do?pageNum='+pageNum+'&pageSize='+pageSize,searchEntity);
    }
    this.save=function (entity) {
        return $http.post('../brand/save.do',entity);
    }
    this.findOne=function (id) {
        return  $http.get('../brand/findOne.do?id='+id);
    }
    this.del=function (ids) {
        return $http.get('../brand/delete.do?ids='+ids);
    }
    this.linkBrand=function () {
        return $http.get('../brand/findLinkedBrand.do');
    }

});